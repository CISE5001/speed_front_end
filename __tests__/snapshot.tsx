/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import Home from '@/pages/moderation';  // Adjust the path accordingly

describe('<Home />', () => {
  const mockArticles = [
    {
      _id: '1',
      dateSubmitted: '2023-09-30',
      articleTitle: 'Test Article 1',
      status: 'Awaiting Approval'
    },
    {
      _id: '2',
      dateSubmitted: '2023-09-29',
      articleTitle: 'Test Article 2',
      status: 'Approved'
    }
  ];

  it('matches snapshot', () => {
    const { asFragment } = render(<Home submittedArticles={mockArticles} />);
    expect(asFragment()).toMatchSnapshot();
  });
});