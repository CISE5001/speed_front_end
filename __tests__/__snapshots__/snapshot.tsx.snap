// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <main>
      <h2>
        Welcome to the Moderation Page
      </h2>
      <h3
        class="description"
      >
        Table of submitted articles
      </h3>
      <br />
      <table>
        <thead>
          <tr>
            <th>
              Date Submitted
            </th>
            <th>
              Article Title
            </th>
            <th>
              Request Status
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              2023-09-30
            </td>
            <td>
              Test Article 1
            </td>
            <td>
              Awaiting Approval
            </td>
            <td>
              <button
                class="approve"
              >
                Approve
              </button>
              <button
                class="reject"
              >
                Reject
              </button>
            </td>
          </tr>
          <tr>
            <td>
              2023-09-29
            </td>
            <td>
              Test Article 2
            </td>
            <td>
              Approved
            </td>
            <td>
              <p>
                No action required
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </main>
  </div>
</DocumentFragment>
`;

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <main>
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="description"
      >
        Get started by editing 
        <code>
          pages/index.js
        </code>
      </p>
      <div
        class="grid"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h3>
            Documentation →
          </h3>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="card"
          href="https://nextjs.org/learn"
        >
          <h3>
            Learn →
          </h3>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="card"
          href="https://github.com/vercel/next.js/tree/canary/examples"
        >
          <h3>
            Examples →
          </h3>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="card"
          href="https://vercel.com/new"
        >
          <h3>
            Deploy →
          </h3>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by
         
        <span
          class="logo"
        >
          <img
            alt="Vercel Logo"
            data-nimg="1"
            decoding="async"
            height="16"
            loading="lazy"
            src="/vercel.svg"
            style="color: transparent;"
            width="72"
          />
        </span>
      </a>
    </footer>
  </div>
</div>
`;
